aten/src/ATen/native/sparse/SparseBlasImpl.cpp

```diff
@@ -2,11 +2,176 @@
 #include <ATen/Config.h>
 #include <ATen/native/mkl/SparseBlasImpl.h>
 #include <ATen/native/sparse/SparseBlasImpl.h>
+#include <ATen/SparseCsrTensorUtils.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Operators.h>
+#else
+#include <ATen/ops/_convert_indices_from_csr_to_coo.h>
+#endif
 
 namespace at {
 namespace native {
 namespace sparse {
 namespace impl {
+
+Tensor& _compressed_row_strided_mm_out(const Tensor& compressed, const Tensor& strided, Tensor& result) {
+  const auto compressed_layout = compressed.layout();
+  const auto compressed_layout_str = at::sparse_csr::layoutToString(compressed_layout);
+
+  // Layout restrictions.
+  TORCH_CHECK(compressed_layout == kSparseCsr || compressed_layout == kSparseBsr,
+      "spmm(", compressed_layout_str, ", Strided): only Csr and Bsr formats are supported for the sparse argument.");
+
+  // Dtype restrictions.
+  TORCH_CHECK(compressed.scalar_type() == strided.scalar_type(),
+      "spmm(", compressed_layout_str, ", Strided): arguments expected to have the same dtype.");
+
+  // Dim restrictions.
+  TORCH_CHECK(compressed.dim() == 2,
+      "spmm(", compressed_layout_str, ", Strided): sparse arguments which are not 2D are not supported.");
```

Comment:
A mental note: this could be relaxed to work with batched inputs without that much of an effort...